conan/tools/cmake/cmakedeps/templates/target_configuration.py

```diff
@@ -174,9 +168,7 @@ def template(self):
 
         ########## AGGREGATED GLOBAL TARGET WITH THE COMPONENTS #####################
         {%- for comp_variable_name, comp_target_name in components_names %}
-
-        target_link_libraries({{root_target_name}} INTERFACE {{ comp_target_name }})
```

Comment:
This is what it was causing the https://github.com/conan-io/conan/issues/11755 error. When using an `add_subdirectory` the target exists but fails to use `target_link_libraries`. This is apparently related to the scope of the target. Creating the targets as INTERFACE GLOBAL worked too but looked more dangerous.